keras_nlp/metrics/perplexity_test.py

```diff
@@ -0,0 +1,338 @@
+        self.assertAlmostEqual(
+            perplexity._aggregate_cross_entropy.numpy(), 2.1148, delta=1e-3
+        )
+        self.assertEqual(perplexity._number_of_samples, 2)
+        self.assertAlmostEqual(perplexity_val.numpy(), 2.8789, delta=1e-3)
+
+        y_true_2 = tf.constant([[2, 0, 0], [1, 2, 3]])
+        y_pred_2 = tf.constant(
+            [
+                [
+                    [2.887, 0.885, 2.973, 2.582],
+                    [0.3838, 2.629, 1.91, 1.802],
+                    [0.2578, 1.081, 1.125, 2.773],
+                ],
+                [
+                    [1.623, 2.784, 0.2109, 2.66],
+                    [2.395, 2.01, 0.252, 1.828],
+                    [0.4482, 2.629, 0.9697, 0.998],
+                ],
+            ]
+        )
+
+        perplexity.update_state(y_true_2, y_pred_2)
+        perplexity_val = perplexity.result()
+        self.assertAlmostEqual(
+            perplexity._aggregate_cross_entropy, 5.5450, delta=1e-3
+        )
+        self.assertEqual(perplexity._number_of_samples, 4)
+        self.assertAlmostEqual(perplexity_val, 3.9998, delta=1e-3)
+
+    def test_merge_state(self):
+        perplexity_1 = Perplexity(from_logits=True, pad_token_id=0)
+        perplexity_2 = Perplexity(from_logits=True, pad_token_id=0)
+
+        y_true_1 = tf.constant([[1, 3, 0], [2, 1, 3]])
+        y_pred_1 = tf.constant(
+            [
+                [
+                    [1.034, 4.797, 2.82, 1.154],
+                    [2.258, 1.591, 1.811, 1.852],
+                    [3.216, 1.037, 0.3662, 2.7],
+                ],
+                [
+                    [1.363, 1.726, 1.898, 2.582],
+                    [1.163, 1.943, 1.761, 1.497],
+                    [2.766, 1.453, 2.61, 2.805],
+                ],
+            ]
+        )
+
+        y_true_2 = tf.constant([[2, 0, 0], [1, 2, 3]])
+        y_pred_2 = tf.constant(
+            [
+                [
+                    [2.887, 0.885, 2.973, 2.582],
+                    [0.3838, 2.629, 1.91, 1.802],
+                    [0.2578, 1.081, 1.125, 2.773],
+                ],
+                [
+                    [1.623, 2.784, 0.2109, 2.66],
+                    [2.395, 2.01, 0.252, 1.828],
+                    [0.4482, 2.629, 0.9697, 0.998],
+                ],
+            ]
+        )
+
+        y_true_3 = tf.constant([[1, 3, 1], [2, 2, 1]])
+        y_pred_3 = tf.constant(
+            [
+                [
+                    [0.7383, 0.882, 0.7295, 2.64],
+                    [0.867, 1.588, 2.291, 0.967],
+                    [0.0908, 1.453, 1.5, 0.0703],
+                ],
+                [
+                    [2.783, 0.9785, 2.664, 0.507],
+                    [0.741, 1.535, 2.16, 2.531],
+                    [2.863, 1.591, 1.403, 0.885],
+                ],
+            ]
+        )
+
+        perplexity_1.update_state(y_true_1, y_pred_1)
+        perplexity_1.update_state(y_true_2, y_pred_2)
+        self.assertAlmostEqual(
```

Comment:
Please avoid directly checking the private members, if we think it is something necessary to check, then we should consider exposing a public interface for it. 

for this case, can we just check the perplexity value?

Reply:
Ah, yes. I just wanted to show that cross_entropy_for_state_1 + cross_entropy_for_state_2 = cross_entropy_for_state_3. But yeah, checking for perplexity will suffice. Changes made!

Reply:
Thanks for fixing! Please also apply similar changes on other test cases, basically we never want to explicitly check private fields, because private fields are subject to change without notice.

Reply:
@chenmoneygithub, done! Thanks!